### DISCUSSION TITLE
Calculate folder size on selection with "Space" key

### DISCUSSION THREAD

**Discussion by nitrostudios** [mikekazakov/nimble-commander#151] [Ideas]
Hello, would it be possible, please, to include option in the setting to be able to calculate folder size on selection with "Space" key?

Thank you very much!

**Comment by mikekazakov** [comment]
Does it need to be specifically Space and not something like Ctrl+Space? The latter is already supported out of the box - simply change the hotkey of the "Calculate Folder Sizes" to this keystroke. The issue with the Space key without any modifiers is that it cannot be a hotkey in terms of MacOS UX. NC circumvents these limitations in a few situations (see the File Panels section in the Hotkey tab), but I'd be a bit cautious about expanding this list much more, because it breaks principles of a typical MacOS UX.

**Comment by nitrostudios** [comment]
That's interesting, since currently I have mapped F3 key as preview and when pressing Space it just selects the item (mark it as red) without calculating the size if it is a folder, I thought it won't be somehow complicated and does not influence the behavior. I know I can use hot key, but it's much faster to use just one key and have both item selected and size calculated. What do you think? Is it a good idea?

**Comment by mikekazakov** [comment]
Hm, by any chance do you have File Panels -> Toggle Selection set to Space?

**Comment by nitrostudios** [comment]
Yes, exactly.

**Comment by mikekazakov** [comment]
Well that's what I referred to - Nimble Commander currently has some bypasses here and there, but there's no overall system that would allow associating any action with a keystroke that macOS doesn't recognise as a valid hotkey.
Expanding the set of "hacked-in actions" in NC doesn't seem like the right forward to me.
I'll move this issue into Discussions.
